syntax = "proto3";
package monitor.proto;
option go_package = ".;proto";

import "net_info.proto";
import "mem_info.proto";
import "cpu_stat.proto";
import "cpu_softirq.proto";
import "cpu_load.proto";

// Message types for overall monitor information
// pass as both parameter and return-value
message MonitorInfo {
    string name = 1; // Monitor instance name
    repeated SoftIrq soft_irq = 4;  // Soft Interrupt statistics
    CpuLoad cpu_load = 5; // CPU load averages
    repeated CpuStat cpu_stat = 6; // CPU statistics
    MemInfo mem_info = 7;  // Memory information
    repeated NetInfo net_info = 8; // Network information
}